pystac/extensions/mgrs.py

```diff
@@ -0,0 +1,249 @@
+        raise ValueError("Invalid MGRS utm zone: must be None or int")
+    if v is not None and v not in UTM_ZONES:
+        raise ValueError(f"Invalid MGRS UTM zone: {v} is not in {UTM_ZONES}")
+    return v
+
+
+class MgrsExtension(
+    PropertiesExtension,
+    ExtensionManagementMixin[Union[pystac.Item, pystac.Collection]],
+):
+    """A concrete implementation of :class:`MgrsExtension` on an :class:`~pystac.Item`
+    that extends the properties of the Item to include properties defined in the
+    :stac-ext:`MGRS Extension <mgrs>`.
+
+    This class should generally not be instantiated directly. Instead, call
+    :meth:`MgrsExtension.ext` on an :class:`~pystac.Item` to extend it.
+
+    .. code-block:: python
+
+       >>> item: pystac.Item = ...
+       >>> proj_ext = MgrsExtension.ext(item)
+    """
+
+    item: pystac.Item
+    """The :class:`~pystac.Item` being extended."""
+
+    properties: Dict[str, Any]
+    """The :class:`~pystac.Item` properties, including extension properties."""
+
+    def __init__(self, item: pystac.Item):
+        self.item = item
+        self.properties = item.properties
+
+    def __repr__(self) -> str:
+        return "<ItemMgrsExtension Item id={}>".format(self.item.id)
+
+    def apply(
+        self,
+        latitude_band: str,
+        grid_square: str,
+        utm_zone: Optional[int] = None,
+    ) -> None:
+        """Applies MGRS extension properties to the extended Item.
+
+        Args:
+            latitude_band : REQUIRED. The latitude band of the Item's centroid.
+            grid_square : REQUIRED. MGRS grid square of the Item's centroid.
+            utm_zone : The UTM Zone of the Item centroid.
+        """
+        self.latitude_band = validated_latitude_band(latitude_band)
+        self.grid_square = validated_grid_square(grid_square)
+        self.utm_zone = validated_utm_zone(utm_zone)
+
+    @property
+    def latitude_band(self) -> Optional[str]:
+        """Get or sets the latitude band of the datasource."""
+        return self._get_property(LATITUDE_BAND_PROP, str)
+
+    @latitude_band.setter
+    def latitude_band(self, v: str) -> None:
+        self._set_property(
+            LATITUDE_BAND_PROP, validated_latitude_band(v), pop_if_none=False
+        )
+
+    @property
+    def grid_square(self) -> Optional[str]:
+        """Get or sets the latitude band of the datasource."""
+        return self._get_property(GRID_SQUARE_PROP, str)
+
+    @grid_square.setter
+    def grid_square(self, v: str) -> None:
+        self._set_property(
+            GRID_SQUARE_PROP, validated_grid_square(v), pop_if_none=False
+        )
+
+    @property
+    def utm_zone(self) -> Optional[int]:
+        """Get or sets the latitude band of the datasource."""
+        return self._get_property(UTM_ZONE_PROP, int)
+
+    @utm_zone.setter
+    def utm_zone(self, v: Optional[int]) -> None:
+        self._set_property(UTM_ZONE_PROP, validated_utm_zone(v), pop_if_none=True)
```

Comment:
This was `pop_if_none=False` in the original, but None is not allowed according to the spec.